catroid/src/main/java/org/catrobat/catroid/utils/HtmlRegexExtractor.java

```diff
@@ -27,19 +27,70 @@
 
 import org.catrobat.catroid.R;
 
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import androidx.annotation.VisibleForTesting;
+
 public class HtmlRegexExtractor {
+
 	private Context context;
+
 	public HtmlRegexExtractor(Context context) {
 		this.context = context;
 	}
-	public void searchKeyword(String search, String text) {
-		int index = text.indexOf(search);
-		if (index >= 0) {
-			ToastUtil.showSuccess(context,
-					R.string.formula_editor_function_regex_html_extractor_found);
+
+	public void searchKeyword(String keyword, String text) {
+		if (findKeyword(keyword, text) == null) {
+			showError();
 		} else {
-			ToastUtil.showError(context,
-					R.string.formula_editor_function_regex_html_extractor_not_found);
+			showSuccess();
+		}
+	}
+
+	private void showSuccess() {
+		ToastUtil.showSuccess(context,
+				R.string.formula_editor_function_regex_html_extractor_found);
+	}
+	private void showError() {
+		ToastUtil.showError(context,
+				R.string.formula_editor_function_regex_html_extractor_not_found);
+	}
+
+	@VisibleForTesting
+	public String findKeyword(String keyword, String text) {
+		if (keyword.equals("")) {
+			throw new IllegalArgumentException("No empty keywords allowed");
+		}
+		if (text.indexOf(keyword) >= 0) {
+			return keyword;
+		} else {
+			return findKeywordWithHtmlBetweenWordsInText(keyword, text);
+		}
+	}
+
+	private String findKeywordWithHtmlBetweenWordsInText(String keyword, String text) {
+		String[] splittedKeyword = keyword.split(" ");
+		String regex = Pattern.quote(splittedKeyword[0]);
+
+		for (int i = 1; i < splittedKeyword.length; i++) {
+			regex += ".*?" + Pattern.quote(splittedKeyword[i]);
```

Comment:
I'm now wondering if that's not too much sometimes. Anything can go in between, after all, also long bunches of visible parts of text. Well, we'll see. I guess in normal cases it will be useful anyway.